device/alpaka/src/clusterization/measurement_sorting_algorithm.cpp

```diff
@@ -29,16 +28,15 @@ measurement_sorting_algorithm::operator()(
         m_copy.get().get_size(measurements_view);
 
     // Sort the measurements in place
-    // TODO: This obviously won't work for HIP, but we don't have a HIP
-    // equivalent of Thrust yet.
-#ifdef ALPAKA_ACC_GPU_CUDA_ENABLED
-    thrust::sort(thrust::device, measurements_view.ptr(),
+#if defined(ALPAKA_ACC_GPU_CUDA_ENABLED) || defined(ALPAKA_ACC_GPU_HIP_ENABLED)
+    auto thrustExecPolicy = thrust::device;
+#else
+    auto thrustExecPolicy = thrust::host;
+#endif
+
+    thrust::sort(thrustExecPolicy, measurements_view.ptr(),
```

Comment:
Attila has already pointed out that this could / should be extracted out to somewhere more general (likely swapping the `alpaka::vecmem_resource` namespace to something more generic, and putting it there).

My preference would be to tackle that in a follow up, just because there is some other slightly larger work that is also needed around making the code generic for all backends (mostly, making the interface for getting vecmem resources truly unified, as currently SYCL is still treated as an edge case with an `ifdef`, as shown in the example updates below).